core/src/main/java/it/pagopa/selfcare/party/registry_proxy/core/NationalRegistriesServiceImpl.java

```diff
@@ -36,6 +40,35 @@ public VerifyLegalResponse verifyLegal(String taxId, String vatNumber) {
         return nationalRegistriesConnector.verifyLegal(taxId, vatNumber);
     }
 
+    @Override
+    public Businesses institutionsByLegalTaxId(String legalTaxId) {
+        log.info("institutionsByLegalTaxId for legalTaxId: {}", MaskDataUtils.maskString(legalTaxId));
+        Businesses response = this.nationalRegistriesConnector.institutionsByLegalTaxId(legalTaxId);
+        if(checkIfResponseIsInfoCamereNotFoundError(response)) {
+            log.info("institutions not found for legalTaxId: {}",MaskDataUtils.maskString(legalTaxId));
+            return createOkResponse(legalTaxId);
+        }
+
+        return response;
+    }
+
+    private boolean checkIfResponseIsInfoCamereNotFoundError(Businesses response) {
```

Comment:
This isn't required for this PR, but... as a general rule, these cases should be handled with a mapping target class (see https://mapstruct.org/documentation/stable/reference/html/#basic-mappings).

Instead of returning a boolean after some manual checks, you may return the entire converted object (i.e. 'MyFullBusinessResponse') that you can safely pass around to other methods.

Furthermore, in that way you automatically add a compile-time check to ensure that the response object returned by 'institutionsByLegalTaxId' is of the expected type ('MyFullBusinessResponse') rather than the unvalidated type ('Business').

Reply:
@flaminiaScarciofolo you can look at an example of mapstruct usage in selfcare onboarding-backend https://github.com/pagopa/selfcare-onboarding-backend/blob/main/web/src/main/java/it/pagopa/selfcare/onboarding/web/model/mapper/OnboardingResourceMapper.java